### DISCUSSION TITLE
Adding Obsidian Site Generator Mode

### DISCUSSION THREAD

**Discussion by RoyiAvital** [srid/neuron#620] [Ideas]
If Neuron will have an _Obsidian Compatible Mode_ it can become the non official [Obsidian](https://obsidian.md/) site generator.

Since the site generation is a paid option for  [Obsidian](https://obsidian.md/) this can also become a paid feature or to the least increase exposure and donations.

I think most features are already implemented in Neuron, the missing parts are:

1. Supporting embedding text from another note into the other (With the `![]` syntax).
2. Supporting embedding files (See https://help.obsidian.md/How+to/Embed+files).
3. Understanding of _Blocks_ in Obsidian (See https://help.obsidian.md/How+to/Link+to+blocks).

Once the above is added 2 more features can be added:

1. Supporting Obsidian syntax for search.
2. Having "Open to the Right" option.

I am writing this as I think Neuron's best feature is the elegant Web Site it generates.
It makes sense to elevate this and generate financial support for the project.

**Comment by srid** [comment]
@RoyiAvital I'm currently working on next-gen version of neuron [here](https://github.com/srid/emabook). I'll check out these aspects of Obsidian to see how we can use them in the new neuron (which incidentally provides an opportunity to try out new things, and break away from the norm of existing neuron). 

(I like Obsidian's search syntax using code blocks; [we should use them](https://github.com/srid/neuron/issues/407#issuecomment-800449103))

**Reply by RoyiAvital** [reply]
I'd rather see on the current Neuron with its known features and look (Though it is a pity its built in search was removed).

This is a fast track to get some financial support.

Also, don't you think moving from one project to another so quickly is the wrong message for users?
After all the slogan of Neuron is being future proof.

**Reply by srid** [reply]
@RoyiAvital The next-gen version will support Neuron's features, and will have a better look - though nothing will prevent you from using neuron's look (as unlike neuron, the next-gen version allows customizing HTML fully). There will be search too, eventually. The "future-proof" part means that, you can use the next-gen version on your existing notebook and it will just work\*.

Once it is released, why will you want to keep using current Neuron despite these improvements?

If you want to provide feedback, you can try the WIP version following instructions [here](https://github.com/srid/emabook#installing-and-using) and drop by in the Matrix channel to discuss (we recently talked about the .yaml feature; specifying custom data on per-Markdown / per-directory basis, which is another cool new feature not possible with neuron).

\* Except for replacing `z:zettels?..` queries with Obsidian-style ones

**Reply by srid** [reply]
As for,

> Also, don't you think moving from one project to another so quickly is the wrong message for users?

There are a few reasons for starting a new project rather than modifying the existing one, I'll go over them in the announcement post. Suffice it to say, since nobody is paying me to develop neuron I do not have to prioritize what some users think over my own appraisal of what's good for the software. :-P Besides, users are welcome to give me specific feedback on what this "wrong message" is, which in turn may tune my appraisal. For instance, you may answer my query above, which might in turn reveal something I've missed so far.

**Reply by RoyiAvital** [reply]
I think you have the best approach of all tools out there.
It is good enough that you deserve to monetize it.

The way I think will be most profitable for you as the developer and us as users (Keeping you on this project and for long time) is by being the unofficial Obsidian static site generator.

I think the template you created for Neuron (Though unfortunately embedded in the code) is almost perfect.
Add to is Tufte style notes (On the sides), open to the right (Of a single note, not independently like in Obsidian) and a graph and it is perfect.

I hope the new engine will have a very similar, or at least an option for  a similar, template as neuron.

I would be happy to try your new application.
But again, It is not available for Windows.

What about embracing the development procedure we saw the other time to have Windows support?
Have you thought about moving the code to Go?

**Reply by srid** [reply]
> Have you thought about moving the code to Go?

LMAO not a chance in hell! 

I realize you have asked for Windows support more than a few times. I don't use Windows. Your best bet is to use [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10) and call it a day. With VSCode on Windows (in addition to [this extension](https://github.com/svsool/vscode-memo) for Zettelkasten) you get [near-native experience](https://code.visualstudio.com/blogs/2019/09/03/wsl2) with WSl2 programs. 

> I think the template you created for Neuron (Though unfortunately embedded in the code) is almost perfect.
Add to is Tufte style notes (On the sides), open to the right (Of a single note, not independently like in Obsidian) and a graph and it is perfect.
> 
> I hope the new engine will have a very similar, or at least an option for a similar, template as neuron.

There is a demo here: https://haskell-kb.srid.ca/ (user-customizable HTML template [here](https://github.com/srid/emabook/blob/d51bac748a7d528204e74db6bef9d37c012e1b2e/docs/templates/_default.tpl)) - you'll notice the sidebar, which is new. I would like to somehow add the up-tree UI from neuron too (not sure how yet) and/or consider a more general graph visualization. Tufte style notes like [footnotes to side](https://rstudio.github.io/tufte/)? Do you have an online demo of 'open to the right' UX?

Now is the most suitable time to provide any and all feedback on next-gen neuron (its UI, features, etc.).

**Reply by RoyiAvital** [reply]
Indeed, Tufte is the idea of footnotes -> side notes.
You may have a look at https://simply-jekyll.netlify.app/posts/how-to-use-simply-jekyll-features-on-your-website for how it is done in Jekyll.

I don't have an example of open to the right but  [Andy Matuschak's Notes Style](https://notes.andymatuschak.org/).
It is also on the demo pages of [Obsidian Publish](https://obsidian.md/publish).
Yet I think it is exaggerated. I think having only a single window to the right / left is enough.
If you happen to implement it, it is important to have the option to close pages (Not in the templates above).

Regarding features, I think copying Obsidian most important features is a must:
1. Embed notes by `![[]]`.
2. Link to any element of a note (Header, Paragraph, etc...).

Probably (2) is hard so even only linking to headers will be great (Also embedding).

UI wise, I think the current UI of Neuron is much better. I also not sure about the tree on the left. If we're after a Digital Garden, it makes sense to only have a search, tags and categories (Categories are _Hard_ like a draw - Can be in a single category, Tags are _Soft_) will do. A graph will be a nice extra (Not a must, certainly not at launch).

Anyhow, as long as it can be adjusted to be like Neuron, I'm perfectly OK with any default.

**Comment by srid** [comment]
Embedding of notes and images are now supported in Emanote: https://emanote.srid.ca/demo/embed

**Reply by RoyiAvital** [reply]
Is there a guide to get the exact look of Neuron on Emanots?

**Reply by srid** [reply]
Not exactly, but similar: https://emanote.srid.ca/demo/neuron-layout

If you want exact look, you will want to write the HTML by overriding the .tpl files, but there is no documentation on that. Basically you can override any file under https://github.com/srid/emanote/tree/master/default by copying to your notebook and modifying them.
